### DISCUSSION TITLE
Mesh Terrain Rendering

### DISCUSSION THREAD

**Discussion by volkanigde** [visualdesigncafe/nature-renderer#19] [Help]
Hello,

How can I use Nature Renderer with a mesh terrain? Produced with 'Gaia' or 'Terrain to Mesh' assets.

Thank you.
Volkan

**Comment by maxplooi** [comment]
Mesh terrains are not supported at the moment. Nature Renderer requires a terrain object to load all the placement data and heightmap, so it does not work with only a mesh.

One workaround is to keep a terrain object in the scene and just disable the renderer of the terrain. That way the terrain data is still available for Nature Renderer and you can use a mesh to render the terrain.

**Reply by chadefranklin** [reply]
I also have a mesh terrain system I would like to use Nature Renderer with and I would like to place my detail objects procedurally. It would be inefficient to create tens of thousands of GameObjects with NatureInstance components. Would it be possible to add a scripting API to pass NatureRenderer a NativeArray of positions / matrices to render a certain prefab?
